apps/omg_api/lib/ethereum_event_listener/core.ex

```diff
@@ -98,21 +108,26 @@ defmodule OMG.API.EthereumEventListener.Core do
     %__MODULE__{state | cached: %{cached_data | data: data ++ new_events}}
   end
 
-  @spec get_events(t(), non_neg_integer) :: {:ok, list(event), list(), t()}
+  @spec get_events(t(), non_neg_integer) :: {:ok, list(event), list(), non_neg_integer, t()}
   def get_events(
         %__MODULE__{
           synced_height_update_key: update_key,
-          block_finality_margin: block_finality_margin,
           cached: %{data: data} = cached_data
         } = state,
         sync_height
       ) do
-    sync = sync_height - block_finality_margin
+    sync = apply_margin(sync_height, state)
     {events, new_data} = Enum.split_while(data, fn %{eth_height: height} -> height <= sync end)
 
-    db_update = [{:put, update_key, sync}]
+    db_update = [{:put, update_key, sync_height}]
```

Comment:
If block_finality_margin will be reduced in the settings, 
holes will be created in the processed events

Reply:
yeah, known issue, but this `margin` shouldn't be touched, by design.

Reply:
but why don't save sync instead of sync_height?

Reply:
because it's different for services with their respective `block_finality_margin`s. If this was saved and then read from DB and sent to `RootChainCoordinator`, it would be different from where it left off.

Also there's services that behave differently, like `BlockGetter`.

Only solution would be to add `block_finality_margin` back in and hope for the best, but that seems  more awkward than the solution prior to the change you've made.

`RootChainCoordinator` was designed to provide a common view of the root chain height, which is admissible to be consumed by services, without taking their specific requirements on finality. Changing that is a deep reorganization and let's not delve into this now (we will later)